core/src/main/java/org/elasticsearch/action/search/RemoteClusterConnection.java

```diff
@@ -373,10 +373,20 @@ void collectRemoteNodes(Iterator<DiscoveryNode> seedNodes,
                             // here we pass on the connection since we can only close it once the sendRequest returns otherwise
                             // due to the async nature (it will return before it's actually sent) this can cause the request to fail
                             // due to an already closed connection.
-                            transportService.sendRequest(connection,
-                                ClusterStateAction.NAME, request, TransportRequestOptions.EMPTY,
+                            ThreadPool threadPool = transportService.getThreadPool();
+                            ThreadContext threadContext = threadPool.getThreadContext();
+                            TransportService.ContextRestoreResponseHandler responseHandler = new TransportService
+                                .ContextRestoreResponseHandler(threadContext
+                                .newRestorableContext(false),
```

Comment:
Should the `ContextRestoreResponseHandler` have a `<ClusterStateResponse`> type parameter?